lcc_lokavaluto_app_connection/models/transaction_rule.py

```diff
@@ -13,5 +14,30 @@ class TransactionRule(models.Model):
     sequence = fields.Integer()
     sender_wallet_domain = fields.Char("Sender Wallet Domain")
     recipient_wallet_domain = fields.Char("Recipient Wallet Domain")
-
     is_transaction_allowed = fields.Boolean("Is Transaction Allowed?")
+
+    recipients_matched_by_rule = fields.Many2many(
+        "res.partner.backend",
+        string="Matched Recipients",
+        compute="_compute_recipients_matched_by_rule",
+        store=False
+    )
+
+    @api.depends("recipient_wallet_domain")
+    def _compute_recipients_matched_by_rule(self):
+        for rule in self:
+            rule.recipients_matched_by_rule = self.env["res.partner.backend"].search(
+                safe_eval(self.recipient_wallet_domain)
+            )
+
+    def recipient_is_allowed_by_rule(self, recipient):
```

Comment:
if x : return True else: return False, is equivalent to return x
    (A and B) or (not A and not B) is equivalent to not (A ^ B) (this is a xor)